### ISSUE TITLE
go test fails on Mac M1

### ISSUE THREAD

**Issue by fasmat** [spacemeshos/smkeys#6]
When running `make deps` and then `make test` on my machine it fails to load the dylibs required:

```shell
$ make test
CGO_CFLAGS="-I/Users/fasmat/workspace/spacemesh/smcli/smkeys/deps" \
        CGO_LDFLAGS="-L/Users/fasmat/workspace/spacemesh/smcli/smkeys/deps" \
        LD_LIBRARY_PATH=/Users/fasmat/workspace/spacemesh/smcli/smkeys/deps \
        go test -v -count 1 -ldflags "-extldflags \"-L/Users/fasmat/workspace/spacemesh/smcli/smkeys/deps -led25519_bip32 -lspacemesh_remote_wallet\"" ./...
dyld[28931]: Library not loaded: /Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib
  Referenced from: <A55A6C4C-5438-3286-9CB2-389E89AF0791> /private/var/folders/4z/mx354sss66ndtlb135lfjnmc0000gn/T/go-build3364499586/b001/bip32.test
  Reason: tried: '/Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib' (no such file), '/Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib' (no such file), '/usr/local/lib/libed25519_bip32.dylib' (no such file), '/usr/lib/libed25519_bip32.dylib' (no such file, not in dyld cache)
FAIL    github.com/spacemeshos/smkeys/bip32     0.184s
dyld[28932]: Library not loaded: /Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib
  Referenced from: <1B6AB72F-EF32-3707-B45D-E08875F5E2F3> /private/var/folders/4z/mx354sss66ndtlb135lfjnmc0000gn/T/go-build3364499586/b156/remote-wallet.test
  Reason: tried: '/Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib' (no such file), '/Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/libed25519_bip32.dylib' (no such file), '/usr/local/lib/libed25519_bip32.dylib' (no such file), '/usr/lib/libed25519_bip32.dylib' (no such file, not in dyld cache)
FAIL    github.com/spacemeshos/smkeys/remote-wallet     0.330s
FAIL
make: *** [test] Error 1
```

**Comment by lrettig** [comment]
I see this issue on Mac too. I think it's not visible in CI because a. we only run tests on linux in CI, and b. the release workflow (for smcli) runs on Mac but doesn't run tests.

**Comment by fasmat** [comment]
The issue is probably fixed by setting rpath settings similar to https://github.com/spacemeshos/post-rs/blob/9f7b5d70ba6ca753f6a0b92bbef8e5cd3fe93480/Cargo.toml#LL52C3-L55C13

It probably works on the CI because there the dylibs are in the folder `/Users/aviv/actions-runner/_work/spacemesh-sdk/spacemesh-sdk/target/release/deps/`.

**Comment by fasmat** [comment]
I updated the makefile to download v1.0.0 of `spacemesh-sdk` and then run the tests. It seems like the artifacts of that release don't contain any binaries? I created an issue: https://github.com/spacemeshos/spacemesh-sdk/issues/12

**Comment by lrettig** [comment]
> I updated the makefile to download v1.0.0 of `spacemesh-sdk` and then run the tests. It seems like the artifacts of that release don't contain any binaries? I created an issue: [spacemeshos/spacemesh-sdk#12](https://github.com/spacemeshos/spacemesh-sdk/issues/12)

Fixed! Use v1.0.1: https://github.com/spacemeshos/spacemesh-sdk/releases/tag/v1.0.1

**Comment by lrettig** [comment]
To expand upon what I said above:

- this isn't causing problems in CI here because CI tests are run only on Linux
- we don't run any release workflow for smkeys
- we do release smcli (including on macOS), which depends on this, but the libraries are statically linked there

It _only_ causes issues for tests, which use dynamic linking - in both this repository and in smcli.
